keanu-python/keanu/vertex/lambda_model.py

```diff
@@ -0,0 +1,55 @@
+from keanu.context import KeanuContext
+from keanu.vertex import Vertex
+from keanu.vertex.vertex_label import VertexLabel
+from keanu.tensor import Tensor
+from keanu.functional import Consumer, Function
+from keanu.vertex.base import JavaObjectWrapper
+from py4j.java_gateway import java_import
+
+
+context = KeanuContext()
+java_import(context.jvm_view(), "io.improbable.keanu.vertices.model.LambdaModelVertex")
+
+def _extract_values(vertices):
+    value_map = {k: JavaObjectWrapper(v.getValue()) for k, v in vertices.items()}
+    return context.to_java_map(value_map)
+
+class LambdaModel(Vertex):
+    def __init__(self, 
+                inputs: map, 
+                executor, 
+                update_value = _extract_values
+            ) -> Vertex:
+        self.executor = executor
+        vertex = context.jvm_view().LambdaModelVertex(
+            LambdaModel.__to_java_map(inputs), 
+            Consumer(self.__execute), 
+            Function(update_value)
+        )
+        super(LambdaModel, self).__init__(vertex)
+
+    @staticmethod
+    def __to_java_map(inputs):
+        unwrapped_inputs = {VertexLabel(k).unwrap(): v for k,v in inputs.items()}
+        return context.to_java_map(unwrapped_inputs)
+
+    def __execute(self, vertices):
+        vertices_wrapped = LambdaModel.__wrap(vertices)
+        self.executor(vertices_wrapped)
+        LambdaModel.__unwrap(vertices_wrapped, vertices)
+
+    @staticmethod
+    def __wrap(vertices):
+        return {k.getUnqualifiedName(): Vertex(v) for k, v in vertices.items()}
```

Comment:
I deliberately left `VertexLabel` out of the Python side of things because I thought it was more Pythonic to recklessly throw strings around 😆 

In all seriousness though, I think Python users would expect not to have the extra overhead of creating additional types. Thoughts @christophernorth ?

Reply:
Sorry - I missed this!  I agree with Mike here - recklessly throwing around strings seems best unless VertexLabels ever got some funky functionality.

Reply:
😄

Reply:
Sounds good